src/test/java/com/ace/alfox/MoveActionTests.java

```diff
@@ -0,0 +1,65 @@
+package com.ace.alfox;
+
+import com.ace.alfox.game.MoveAction;
+import com.ace.alfox.game.interfaces.IAction;
+import com.ace.alfox.game.models.Player;
+import com.ace.alfox.lib.ActionResult;
+import com.ace.alfox.lib.Database;
+import com.ace.alfox.lib.Vector2;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class MoveActionTests {
+    Player player;
+    IAction move;
+    static Database db;
+
+    @BeforeClass
+    public static void createConnection() {
+        db = new Database("memory");
+    }
+
+    @Before
+    public void setUp() {
+        player = new Player();
+        move = new MoveAction(db);
+    }
+
+    @Test
+    public void Move() {
```

Comment:
Would love to see a test for moving every direction, but not a blocker. This is fine for now!

Capitals on method names seems super weird to me! Though test names I'm used to are `test_whatIsHappening_expectedResult` which is super verbose and everyone I talk to seems to hate it. Though it was great at Yelp; where you viewed test results knew about the format and would print it real nice. Made understanding what broke a lot faster usually.

Reply:
haha, some C# slipping through. My bad.

Yea I'm fine with that naming scheme, tests are suppose to be verbose and fail loudly. I haven't seen the `_expectedResult` pattern before but can definitely see the appeal. I'm not scared of long function names :P

Reply:
Ah yeah I forgot that about C# :D